packages/opentelemetry-context-base/src/index.ts

```diff
@@ -14,6 +14,9 @@
  * limitations under the License.
  */
 
-export * from './types';
-export * from './context';
 export * from './NoopContextManager';
+export * from './types';
+import { BaseContext } from './context';
+import { Context } from './types';
+export { createContextKey } from './context';
+export const ROOT_CONTEXT: Context = BaseContext.ROOT_CONTEXT;
```

Comment:
I find this a bit confusing why creating an extra variable `ROOT_CONTEXT` if it exists on BaseContext and is exactly the same. Then in one place `BaseContext.ROOT_CONTEXT` is used instead of ROOT_CONTEXT, is this on purpose or mistake? (in NoopContextManager.ts).

I'm not sure if `BaseContext` needs to have `ROOT_CONTEXT`, can we just always use `ROOT_CONTEXT` and then remove static `ROOT_CONTEXT` from `BaseContext` so people will always use only one `ROOT_CONTEXT` that is define on `index.ts`.

```
export const ROOT_CONTEXT: Context = new BaseContext();
```

Reply:
The constructor of `BaseContext` is private so it can't be constructed manually.

I just realized though that we can make the constructor public again though since the class is not actually exported so there is no risk anymore.

Reply:
See [`678e9b5` (#1515)](https://github.com/open-telemetry/opentelemetry-js/pull/1515/commits/678e9b56036c232ac0d6ac5be3cd44e0a6307818)